src/utils/EventEmitter.class.js

```diff
@@ -0,0 +1,40 @@
+// from https://gist.github.com/mudge/5830382#gistcomment-2691957
+/** Enables an instance of a class to emit its own events (and be listened to) */
+export default class EventEmitter {
+    constructor() {
+        this.events = {}
+    }
+
+    _getEventListByName(eventName) {
+        if (typeof this.events[eventName] === 'undefined') {
+            this.events[eventName] = new Set()
+        }
+        return this.events[eventName]
+    }
+
+    on(eventName, fn) {
+        this._getEventListByName(eventName).add(fn)
+    }
+
+    once(eventName, fn) {
+        const self = this
+
+        const onceFn = function (...args) {
+            self.removeListener(eventName, onceFn)
+            fn.apply(self, args)
+        }
```

Comment:
An arrow function would be nice here.

Reply:
you mean so that I could use `this` inside instead of `self`?

Reply:
Yeah. I guess the code is copy/pasted. And it's ok to keep close to the original.
Just something I stumbled over while reading it :smiley:

Reply:
> Yeah. I guess the code is copy/pasted.

![Nope](https://media3.giphy.com/media/Atc9QCyWLGHgLZhHDp/giphy.gif?cid=ecf05e47ty6tdre0xkitb5yq54t4q4j150swx6t72mg12otn&rid=giphy.gif&ct=g)

Reply:
:laughing: